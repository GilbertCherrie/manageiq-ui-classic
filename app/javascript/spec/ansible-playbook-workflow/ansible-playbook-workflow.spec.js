import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import AnsiblePlaybookWorkflow from '../../components/ansible-playbook-workflow';
import { ansiblePlaybookData } from './ansible-playbook-workflow.data';
import NotificationMessage from '../../components/notification-message';

describe('AnsiblePlaybookWorkflow component', () => {
  it('should render the AnsiblePlaybookWorkflow with payload', () => {
    const wrapper = shallow(<AnsiblePlaybookWorkflow
      payload={ansiblePlaybookData.payload}
      payloadType={ansiblePlaybookData.payloadType}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render the AnsiblePlaybookWorkflow without payload and display a notification', () => {
    const wrapper = mount(<AnsiblePlaybookWorkflow
      payload={undefined}
      payloadType={ansiblePlaybookData.payloadType}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(NotificationMessage)).toHaveLength(1);
  });
});
