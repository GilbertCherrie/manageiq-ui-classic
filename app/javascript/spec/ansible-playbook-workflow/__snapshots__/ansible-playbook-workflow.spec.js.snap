// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsiblePlaybookWorkflow component should render the AnsiblePlaybookWorkflow with payload 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12 col-lg-6"
  >
    <Tabs
      className="miq_custom_tabs"
      scrollDebounceWait={150}
      scrollIntoView={true}
      selected={0}
      selectionMode="automatic"
      type="default"
    >
      <Tab
        key="tab_text"
        label="Text"
        onClick={[Function]}
        onKeyDown={[Function]}
        selected={false}
      >
        <Controlled
          className="miq-codemirror ansible-playbook-workflow-payload"
          options={
            Object {
              "autoCloseBrackets": true,
              "gutters": Array [
                "CodeMirror-lint-markers",
              ],
              "lineNumbers": true,
              "lineWrapping": true,
              "lint": true,
              "mode": "yaml",
              "styleActiveLine": true,
              "theme": "eclipse",
            }
          }
          value="
  ---
  - name: Provision instances
    hosts: localhost
    connection: local
    gather_facts: False
    # load AWS variables from this group vars file
    vars_files:
      - group_vars/all
    tasks:
      - name: Launch instances
        ec2:
        access_key: \\"{{ ec2_access_key }}\\"
        secret_key: \\"{{ ec2_secret_key }}\\"
        keypair: \\"{{ ec2_keypair }}\\"
        group: \\"{{ ec2_security_group }}\\"
        type: \\"{{ ec2_instance_type }}\\"
        image: \\"{{ ec2_image }}\\"
        region: \\"{{ ec2_region }}\\"
        instance_tags: \\"{'ansible_group':'jboss', 'type':'{{ ec2_instance_type }}', 'group':'{{ ec2_security_group }}', 'Name':'demo_''{{ tower_user_name }}'}\\"
        count: \\"{{ ec2_instance_count }}\\"
        wait: true
        register: ec2
      - name: Wait for SSH to come up
        wait_for:
        host: \\"{{ item.public_dns_name }}\\"
        port: 22
        delay: 60
        timeout: 320
        state: started
        with_items: \\"{{ ec2.instances }}\\""
        />
      </Tab>
    </Tabs>
  </div>
</div>
`;

exports[`AnsiblePlaybookWorkflow component should render the AnsiblePlaybookWorkflow without payload and display a notification 1`] = `
<AnsiblePlaybookWorkflow
  payloadType="yaml"
>
  <div
    className="row"
  >
    <div
      className="col-md-12 col-lg-6"
    >
      <NotificationMessage
        message="Payload is not available."
        type="info"
      >
        <div
          className="miq-notification-message-container alert alert-info"
        >
          <span
            className="pficon pficon-info"
          />
          <strong>
            Payload is not available.
          </strong>
        </div>
      </NotificationMessage>
    </div>
  </div>
</AnsiblePlaybookWorkflow>
`;
